### ISSUE TITLE
[libc++] std::isfinite() does not accept convertible-to-float

### ISSUE THREAD

**Issue by AngryLoki** [llvm/llvm-project#98816]
The following code works with libstdc++, fails with libc++:
```c++
#include <cmath>

struct X {
    operator float();
};

bool test(X value) {
    return std::isfinite(value);
}
```

Conformance viewer: https://godbolt.org/z/nK9sofToP

Long time ago probably the same issue was reported in https://github.com/llvm/llvm-project/issues/23081 and closed as a duplicate of https://github.com/llvm/llvm-project/issues/18592 . However the fix was applied only for `isnan` and `isinf` - see https://github.com/llvm/llvm-project/blob/fe7fc54403e17b17e366adaab75256b2269fc204/libcxx/include/__math/traits.h#L97-L110

Could you apply the same fix for `std::isfinite`?

**Comment by vortex73** [comment]
I can look into this issue. Please assign.
